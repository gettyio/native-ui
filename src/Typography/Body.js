import styled from 'styled-components/native'

const Body = styled.Text`
  font-size: 23px;
  font-weight: normal;
  line-height: 29px;
  letter-spacing: -0.384;
`

export default Body
