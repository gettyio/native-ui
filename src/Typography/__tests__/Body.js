/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Body from '../Body'

describe('Body component', () => {
  it('should render without issue', () => {
    const component = shallow(<Body>Body text.</Body>)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should have default props', () => {
    const component = shallow(<Body>Body text.</Body>)

    expect(toJson(component)).toMatchSnapshot()
    expect(toJson(component)).toHaveStyleRule('font-weight', 'normal')
    expect(toJson(component)).toHaveStyleRule('font-size', 23)
    expect(toJson(component)).toHaveStyleRule('line-height', 29)
    expect(toJson(component)).toHaveStyleRule('letter-spacing', -0.384)
  })
})
