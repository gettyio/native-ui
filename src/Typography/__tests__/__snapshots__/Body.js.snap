// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body component should have default props 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 23,
        "fontWeight": "normal",
        "letterSpacing": -0.384,
        "lineHeight": 29,
      },
      undefined,
    ]
  }
>
  Body text.
</Text>
`;

exports[`Body component should render without issue 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 23,
        "fontWeight": "normal",
        "letterSpacing": -0.384,
        "lineHeight": 29,
      },
      undefined,
    ]
  }
>
  Body text.
</Text>
`;
